### PR TITLE
[ASR Pipe] Update init to set model type and subsequently call parent init method

### LINKED ISSUE #28162: save_pretrained no longer works for AutomaticSpeechRecognitionPipeline
### System Info

transformers-4.37.0.dev0

### Who can help?

@Narsil 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```python
from transformers import pipeline

asr_pipeline = pipeline('automatic-speech-recognition',
                        model="openai/whisper-tiny")

asr_pipeline.save_pretrained("pipeline_save")
```
Gives this error:
```
Traceback (most recent call last):
  File "/Users/bruno/testing.py", line 6, in <module>
    asr_pipeline.save_pretrained("pipeline_save")
  File "/Users/bruno/venv/lib/python3.11/site-packages/transformers/pipelines/base.py", line 883, in save_pretrained
    if self.image_processor is not None:
       ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'AutomaticSpeechRecognitionPipeline' object has no attribute 'image_processor'
```

### Expected behavior

The pipeline should be saved.

save_pretrained a pipeline is used by BentoML, as a result versions of transformers newer than 4.32.1 cannot be used to serve a AutomaticSpeechRecognitionPipeline with bentoML.

https://github.com/bentoml/BentoML/issues/4339

### PR DESCRIPTION
**Pull request by sanchit-gandhi**
# What does this PR do?

Fixes #28162 by overriding the init method of the ASR pipeline class. We first set the model type, then call the parent init method.

### CODE REVIEW

File: src/transformers/pipelines/automatic_speech_recognition.py [hunk 1]
```diff
@@ -203,6 +204,7 @@ def __init__(
         feature_extractor: Union["SequenceFeatureExtractor", str] = None,
         tokenizer: Optional[PreTrainedTokenizer] = None,
         decoder: Optional[Union["BeamSearchDecoderCTC", str]] = None,
+        image_processor: Optional["BaseImageProcessor"] = None,
```
Comment (amyeroberts) [inline src/transformers/pipelines/automatic_speech_recognition.py:207]: Do you need to add it as an argument here? Couldn't you just set `self.image_processor = None` directly below?

Comment (sanchit-gandhi) [reply src/transformers/pipelines/automatic_speech_recognition.py:207]: I could, but chose to keep the same signature at the other pipeline init's. Happy to do it either way tbh

**Comment by amyeroberts** [comment]
The problem with the current solution is that it's a patch over a larger issue: the ASR pipeline is a child of `Pipeline` but doesn't share the `__init__`. So, if there's any new logic added to `Pipeline` which relies on attributes being set or new kwargs this will still fail. 

It also doesn't make sense for this class to accept an image processor. 

The simplest solution (but doesn't address the divergence here) would be making certain attributes - `tokenizer`, `feature_extractor`, `image_processor` - to class attributes with default `None`. This way, any new pipeline which is similar to ASR will already have these required objects without having to handle / set them. 

A better solution is to update the `__init__` so that `super().__init__(*args, **kwargs)` is called at some point either modifying the kwargs before, setting self.type where necessary or adding any additional logic after.

**Comment by sanchit-gandhi** [comment]
Thanks for your comments. I agree that calling the parent `__init__` method is a more elegant solution. Resolved in https://github.com/huggingface/transformers/pull/28486/commits/e942642024aff014d5e7cd739ce7e5eb85abbfc7

**Review by amyeroberts — APPROVED** [review]
Beautiful 🤩  - thanks for iterating! 

For the failing tests on `natten`, rebasing on `main` should resolve

**Comment by amyeroberts** [comment]
@sanchit-gandhi Before merging - could you update the title to reflect the change in fix applied?

**Pull request merged**